django/db/migrations/state.py

```diff
@@ -579,11 +580,19 @@ def render(self, apps):
         body.update(self.construct_managers())
 
         # Then, make a Model object (apps.register_model is called in __new__)
-        return type(
-            str(self.name),
-            bases,
-            body,
-        )
+        try:
+            return type(
```

Comment:
Repeating would be quite rare situation, only for classes with multiple metaclasses, so defining new value used only once in 98% of time would be slower in the end.